python/ray/serve/tests/test_standalone.py

```diff
@@ -159,5 +160,80 @@ def test_middleware():
     ray.shutdown()
 
 
+def test_http_proxy_fail_loudly():
+    # Test that if the http server fail to start, serve.start should fail.
+    with pytest.raises(socket.gaierror):
+        serve.start(http_options={"host": "bad.ip.address"})
+
+    ray.shutdown()
+
+
+def test_no_http():
```

Comment:
NoServer and HeadOnly tests were added. EveryNode is reusing the previous `test_multiple_routers` in the same file